src/Template.php

```diff
@@ -297,8 +297,14 @@ public function getBlockNames(array $context, array $blocks = [])
     }
 
     /**
+     * @param string|TemplateWrapper|array $template
+     * @param string                       $templateName
+     * @param int                          $line
+     * @param int                          $index
```

Comment:
```suggestion
     * @param string|null                   $templateName
     * @param int|null                      $line
     * @param int|null                      $index
```

Note that you can also add native parameter types safely most of the time.